### ISSUE TITLE
Not possible to recover files with Excel v. 2110

### ISSUE THREAD

**Issue by Cichel** [salvois/LargeXlsx#5]
Hi @salvois 
Your library worked great with older excel version, but today I updated it and right now I can't open any of the xlsx files created with LargeXlsx. When trying the option "Open and repair" I'm getting "The file is corrupt and cannot be opened" error. Could take a look at this?

Excel:
`Microsoft® Excel® for Microsoft 365 MSO (Version 2110 Build 16.0.14527.20270) 64-bit `

.Net
```
dotnet --info   
.NET SDK (reflecting any global.json):
 Version:   5.0.402
 Commit:    e9d3381880

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.19043
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\5.0.402\

Host (useful for support):
  Version: 5.0.11
  Commit:  f431858f8b

.NET SDKs installed:
  5.0.402 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 5.0.11 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 5.0.11 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 5.0.11 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
```

**Comment by Cichel** [comment]
Hi!
My colleague suggested to change extension of the "xlsx" file to "zip", decompress, compress to "zip" once more and change the extension back to the "xlsx". It worked - the newly compressed file doesn't give any warnings and the data is fully accessible.

**Comment by tacin** [comment]
I suggest to use build-in System.IO.Compression instead of SharpCompress - it seems that is the solution of the problem.

**Comment by Cichel** [comment]
Hi @tacin
Thanks for the solution!

**Comment by salvois** [comment]
Hi @Cichel ,
thanks for reporting this. I don't have that specific Excel version to try to reproduce the problem, but I'm afraid it could be due to different interpretations of the ZIP format.
Have you tried enabling ZIP64? Does it makes any difference?

As far as System.IO.Compression is concerned, last time I checked it could not be used with LargeXlsx because of https://github.com/dotnet/runtime/issues/1544
That was the primary reason to use SharpCompress.
@tacin do you have different information about that?

Thanks,
   Salvo

**Comment by tacin** [comment]
@salvois, you don't need to use Update mode in ZipArchive - Create is enough so you don't have to care about this issue.
I forked your code, changed ZIP library to ZipArchive and it worked well for ~1mln rows without using memory.

**Comment by salvois** [comment]
Hi @tacin ,
thanks for your suggestion. I've tried replacing SharpCompress with System.IO.Compression and in Create mode ZipArchive does not show the memory problem, that's very good news.

However, I've noticed that for very big files (uncompressed size greater than 4 GB) the versions of both Excel and LibreOffice I have fail to open the file, the former being able to repair it, the latter refuses it altogether. This is similar to the problem I experienced with ZIP64 in SharpCompress before I submitted a patch for it.

Do you have any insight for this? Maybe I am misusing the library.
This problem would prevent me to release a version of LargeXlsx not relying on SharpCompress.

Thanks,
   Salvo

**Comment by tacin** [comment]
Hi @salvois,
Unfortunately System.IO.Compression doesn't support zip64 so it can't compress files greater that 4 GB.
However do you have idea why Excel always asks for repairing xlsx (`Microsoft® Excel® for Microsoft 365 MSO (Version 2110 Build 16.0.13801.21050 32-bit`) or even doesn't open at all (`Microsoft® Excel® for Microsoft 365 MSO (Version 2102 Build 16.0.14527.20270) 64-bit`) if I'm using your library with this code:
```
[HttpGet]
public void DownloadExcel()
{
    Response.ContentType = "application/octet-stream";
    Download(Response.Body);
}

public void Download(Stream stream)
{
    using (var xlsxWriter = new XlsxWriter(stream))
    {
        xlsxWriter
            .BeginWorksheet("Sheet 1")
            .BeginRow().Write("Name").Write("Location").Write("Height (m)")
            .BeginRow().Write("Kingda Ka").Write("Six Flags Great Adventure").Write(139)
            .BeginRow().Write("Top Thrill Dragster").Write("Cedar Point").Write(130)
            .BeginRow().Write("Superman: Escape from Krypton").Write("Six Flags Magic Mountain").Write(126);
    }
}
```
but using `System.IO.Compression` instead of `SharpCompress` then it works fine?

Thanks,
tacin

**Comment by salvois** [comment]
Hi @tacin, @Cichel ,
unfortunately I have no idea why using System.IO.Compression.ZipArchive rather than SharpCompress solves the issue on Excel 2110. I think there may be subtle differences in how they save ZIP metadata, and Excel has become picky about them. Maybe a comparison of the binary files produced by both ZipArchive and SharpCompress may help understanding the issue, but it would be time consuming and I don't know when I could try it by now, sorry.

Looks like System.IO.Compression.ZipArchive does support ZIP64 (https://github.com/dotnet/dotnet-api-docs/pull/7333), but there still is something about it Excel doesn't like.

For the moment, I have released LargeXlsx version 1.3.1 which updates its dependencies, especially SharpCompress to its last version. It would be very helpful if you can give it a try, but I'm not too confident about it because LibreOffice 7.2 already fails to open the Zip64Huge example on my setup.

Thanks,
   Salvo

**Comment by salvois** [comment]
Closing due to inactivity. Please open a new issue if the problem persists.
